source/particle/world.cc

```diff
@@ -830,7 +830,7 @@ namespace aspect
       MPI_Waitall(2*n_neighbors,&n_requests[0],MPI_STATUSES_IGNORE);
 
       // Determine how many particles and data we will receive
-      int total_recv_data = 0;
+      unsigned int total_recv_data = 0;
       for (unsigned int neighbor_id=0; neighbor_id<n_neighbors; ++neighbor_id)
         {
           recv_offsets[neighbor_id] = total_recv_data;
```

Comment:
I think the correct patch is as suggested here (make the count an unsigned, which it really is) and then if there is a follow-up warning in line 866, make sure the lhs of the comparison is cast to an `unsigned` data type (because it *must* be positive or zero).